backend/frontend/tests/test_page_users_and_teams_nav.py

```diff
@@ -121,3 +122,112 @@ def test_user_sees_teams_that_match_the_search_term(self):
+            email="john@doe.email"
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.user.set_password(self.user.email)
+        self.user.save()
+        
+        self.superuser = UserFactory(is_superuser=True)
+        self._setup_user_with_github()
+        self._setup_user_without_github()
+        
+    def _goto_users_page(self):
+        url = self.reverse_url("users_and_teams_nav")
+        self.page.goto(url)
+        self.page.wait_for_load_state()
+    
+
+    def test_all_users_show_up_onload(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        for user in [self.user, self.user_without_github, self.user_with_github]:
+            expect(users_div).to_contain_text(user.get_full_name())
+    
+    def test_superusers_do_not_appear(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        expect(users_div).not_to_contain_text(self.superuser.get_full_name())
+
+    def test_load_more_button_appears(self):
+        users = [UserFactory() for _ in range(21)]
+        
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+
+        expect(users_div).to_contain_text("Load more")
+    
+    def test_users_can_search_github_name(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        search_input = self.page.locator("#user-search-term-box")
+        search_input.press_sequentially("samboha_codes")
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        expect(users_div).to_contain_text("Sam Boha")
+    
+    def test_users_can_search_email(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        search_input = self.page.locator("#user-search-term-box")
+        search_input.press_sequentially("sam@rp")
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        expect(users_div).to_contain_text("Sam Boha")
+    
+    def test_users_can_search_first_name(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        search_input = self.page.locator("#user-search-term-box")
+        search_input.press_sequentially("JohN")
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        expect(users_div).to_contain_text("John Doe")
+    
+    def test_users_can_search_last_name(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        search_input = self.page.locator("#user-search-term-box")
+        search_input.press_sequentially("Doe")
```

Comment:
We should have a search function that takes in a string and returns a database query then test that as a normal function instead of using playwright for everything. 

Remember the testing pyramid. Tests should be fast and specific. Favor unit tests over integration tests.